src/crimson/net/Protocol.h

```diff
@@ -34,7 +34,8 @@ class Protocol {
     // seastar::parallel_for_each(). otherwise, we could erase a connection in
     // the container when seastar::parallel_for_each() is still iterating in
     // it. that'd lead to a segfault.
-    return seastar::yield().then([this, dispatch_reset] {
+    return seastar::yield(
+    ).then([this, dispatch_reset, conn_ref = conn.shared_from_this()] {
```

Comment:
i don't really understand this change. what was the root cause? is it:

1. `proto.close()` gets called. and it yields.
2. the messenger closes the connection. and the connection is destroyed.
3. the continuation `proto.close()` gets scheduled. but nobody informs proto that the conn was closed and freed. so it ends up trying to dereference the freed `conn`.

am i right?

Reply:
3. proto is not aware that the conn (and also the proto) was destroyed. so it ends up accessing its freed member https://github.com/ceph/ceph/blob/dea4ec6d9f736a95675701e46ddc8d5e6a86f42c/src/crimson/net/Protocol.cc#L38, which is detected by the AddressSanitizer.